srtp/srtp.c

```diff
@@ -4598,45 +4651,62 @@ srtp_err_status_t stream_get_protect_trailer_length(srtp_stream_ctx_t *stream,
     return srtp_err_status_ok;
 }
 
+struct get_protect_trailer_length_data {
+    int valid;
+    uint32_t length;
+    uint32_t is_rtp;
+    uint32_t use_mki;
+    uint32_t mki_index;
+};
+
+static int get_protect_trailer_length_cb(srtp_stream_t stream, void *raw_data)
+{
+    struct get_protect_trailer_length_data *data =
+        (struct get_protect_trailer_length_data *)raw_data;
+    uint32_t temp_length;
+
+    data->valid = 1;
+    if (stream_get_protect_trailer_length(stream, data->is_rtp, data->use_mki,
+                                          data->mki_index,
+                                          &temp_length) == srtp_err_status_ok) {
+        if (temp_length > data->length) {
+            data->length = temp_length;
+        }
+    }
```

Comment:
```suggestion
    if (stream_get_protect_trailer_length(stream, data->is_rtp, data->use_mki,
                                          data->mki_index,
                                          &temp_length) == srtp_err_status_ok) {
        data->valid = 1;
        if (temp_length > data->length) {
            data->length = temp_length;
        }
    }
```

Reply:
This just moves the assignment to `data->valid` within the `if`, so that `valid` indicates that you got data from at least one stream.